runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -1414,6 +1415,7 @@ J9::ValuePropagation::constrainRecognizedMethod(TR::Node *node)
                }
             break;
             }
+#endif /* defined(J9VM_OPT_METHOD_HANDLE) */
          case TR::java_lang_invoke_DirectHandle_nullCheckIfRequired:
```

Comment:
Can this also be wrapped under `J9VM_OPT_METHOD_HANDLE`?
`j.l.i.DirectHandle` is also a OpenJ9 specific class used for MH impl, I believe there are many other cases of such constants used JIT.

I did a quick search and found list in https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/codegen/J9RecognizedMethodsEnum.hpp#L955-L1086

Maybe this should be done in a separate PR to ifdef all OpenJ9 MH impl with `J9VM_OPT_METHOD_HANDLE`

Reply:
Yes, I think that cleanup should be done: `java_lang_invoke_DirectHandle_nullCheckIfRequired` should not even be declared if `J9VM_OPT_METHOD_HANDLE` is not defined. Changes like that should be deferred to separate pull requests.